src/Features/CSharp/Portable/ConvertLinq/ConvertForEachToLinqQuery/AbstractConverter.cs

```diff
@@ -183,6 +183,15 @@ private static QueryClauseSyntax CreateQueryClause(ExtendedSyntaxNode node)
             //      OR
             //   c1.SelectMany(n1 => ...
             //
+
+            // Avoid `.Select(x => x)`
+            if (lambdaBody is IdentifierNameSyntax identifier &&
+                identifier.Identifier.ValueText == forEachStatement.Identifier.ValueText &&
+                receiverForInvocation is InvocationExpressionSyntax receiverInvocationExpression)
+            {
+                return receiverInvocationExpression;
+            }
+
             var invokedMethodName = !hasForEachChild ? nameof(Enumerable.Select) : nameof(Enumerable.SelectMany);
```

Comment:
consider moving this above, and tehn only doing your check if `invokedMethodName` is actually `Select`.

Reply:
Done, thanks.